src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataReaderTest/DataReaderTest.cs

```diff
@@ -165,9 +156,27 @@ public static void CollatedDataReaderTest()
             }
             finally
             {
-                cmd.CommandText = $"DROP DATABASE {databaseName}";
+                DataTestUtility.DropDatabase(con, dbName);
+            }
+        }
+
+        private static void CreateCollatedDB(SqlConnection con,string databaseName)
+        {
+            try
+            {
+                if (con.State != ConnectionState.Open)
+                {
+                    con.Open();
+                }
```

Comment:
Beign more specific.
```suggestion
                if (con.State ==ConnectionState.Connecting)
                {
                    Thread.Sleep(300);
                }
                else if (con.State == ConnectionState.Closed || con.State == ConnectionState.Broken)
                {
                    con.Open();
                }
```